zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/authorization/DbAuthorizationState.java

```diff
@@ -109,36 +90,18 @@ public void create(final long authorizationKey, final AuthorizationRecord author
     permissionsColumnFamily.upsert(ownerTypeOwnerIdAndResourceType, permissions);
   }
 
-  @Override
-  public void removePermission(
-      final AuthorizationOwnerType ownerType,
-      final String ownerId,
-      final AuthorizationResourceType resourceType,
-      final PermissionType permissionType,
-      final Set<String> resourceIds) {
-    this.ownerType.wrapString(ownerType.name());
-    this.ownerId.wrapString(ownerId);
-    this.resourceType.wrapString(resourceType.name());
-
-    final var permissions = permissionsColumnFamily.get(ownerTypeOwnerIdAndResourceType);
-    permissions.removeResourceIdentifiers(permissionType, resourceIds);
-
-    if (permissions.isEmpty()) {
-      permissionsColumnFamily.deleteExisting(ownerTypeOwnerIdAndResourceType);
-    } else {
-      permissionsColumnFamily.update(ownerTypeOwnerIdAndResourceType, permissions);
-    }
-  }
-
   @Override
   public void deleteAuthorizationsByOwnerTypeAndIdPrefix(
       final AuthorizationOwnerType ownerType, final String ownerId) {
     this.ownerType.wrapString(ownerType.name());
     this.ownerId.wrapString(ownerId);
-    permissionsColumnFamily.whileEqualPrefix(
-        ownerTypeAndOwnerId,
-        (compositeKey, permissions) -> {
-          permissionsColumnFamily.deleteExisting(compositeKey);
+
+    authorizationByAuthorizationKeyColumnFamily.forEach(
```

Comment:
💭 Let me know what you think about this

💭 Another approach is to introduce a column family to keep track of the owner (ownerId + ownerType) and all its authorizations (authorizationKeys)